packit/cli/utils.py

```diff
@@ -88,22 +89,98 @@ def covered_func(config=None, *args, **kwargs):
         return decorator_cover(_func)
 
 
+def iterate_packages(func):
+    """
+    Decorator for dealing with sub-packages in a package (Monorepo) configuration
+
+    * if packages are specified as an option in CLI then
+      call decorated function just for them
+    * if packages are not specified as an option in CLI but
+      there are multiple packages in the configuration
+      then call the decorated function for all of them
+    * if there is just one package in the configuration
+      then call the decorated function just once
+
+    This method (iterate_packages) **has not** `package_config` key
+    in its kwargs, it has `packages`, but calls a method
+    (func) who needs a `package_config` key and not `packages`!
+    """
+
+    @functools.wraps(func)
+    def covered_func(*args, **kwargs):
+        path_or_url = kwargs["path_or_url"]
+        config = kwargs["config"]
+        decorated_func_kwargs = kwargs.copy()
+        del decorated_func_kwargs["packages"]
+        packages_config: MultiplePackages = get_local_package_config(
+            path_or_url.working_dir,
+            repo_name=path_or_url.repo_name,
+            try_local_dir_last=True,
+            package_config_path=config.package_config_path,
+        )
+        if "packages" in kwargs and kwargs["packages"]:
+            for package in kwargs["packages"].split(","):
+                decorated_func_kwargs["config"] = copy.deepcopy(
+                    config
+                )  # reset working variables like srpm_path
+                decorated_func_kwargs["package_config"] = packages_config.packages[
+                    package
+                ]
+                func(*args, **decorated_func_kwargs)
+        elif hasattr(packages_config, "packages"):
+            for _, package_config in packages_config.packages.items():
+                decorated_func_kwargs["config"] = copy.deepcopy(
+                    config
+                )  # reset working variables like srpm_path
+                decorated_func_kwargs["package_config"] = package_config
+                func(*args, **decorated_func_kwargs)
+        else:
+            decorated_func_kwargs["package_config"] = packages_config
+            func(*args, **decorated_func_kwargs)
+
+    return covered_func
+
+
+def fix_workdir(
```

Comment:
Can you please use a more descriptive name? Something like `checkout_package_workdir` or similar... (But thanks for the docstring. That helps a lot!)